recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -75,6 +146,12 @@ def validate(self):
         if self.settings.os != "Linux" and self.options.shared:
             raise ConanInvalidConfiguration(f"{self.ref} supports building shared libraries only on Linux")
 
+        if self.options.with_otlp_grpc and not self.options.with_otlp:
+            raise ConanInvalidConfiguration("Option 'with_otlp_grpc' requires 'with_otlp'")
+
+        if self.options.with_otlp_http and not self.options.with_otlp:
+            raise ConanInvalidConfiguration("Option 'with_otlp_http' requires 'with otlp'")
```

Comment:
These should be deleted in configure if the requirement option is false too

Reply:
Done I think, if I understood correctly.